src/scalingengine/server/scaling_handler.go

```diff
@@ -110,11 +151,82 @@ func (h *ScalingHandler) ComputeNewInstances(currentInstances int, adjustment st
 		newInstances = int(step) + currentInstances
 	}
 
-	if newInstances < instanceMin {
-		newInstances = instanceMin
-	} else if newInstances > instanceMax {
-		newInstances = instanceMax
+	return newInstances, nil
+}
+
+func (h *ScalingHandler) GetScalingHistories(w http.ResponseWriter, r *http.Request, vars map[string]string) {
+	appId := vars["appid"]
+	startParam := r.URL.Query()["start"]
+	endParam := r.URL.Query()["end"]
+	h.logger.Debug("get-scaling-histories", lager.Data{"appId": appId, "start": startParam, "end": endParam})
+
+	var err error
+	start := int64(0)
+	end := int64(-1)
+
+	if len(startParam) == 1 {
+		start, err = strconv.ParseInt(startParam[0], 10, 64)
+		if err != nil {
+			h.logger.Error("get-scaling-histories-parse-start-time", err, lager.Data{"start": startParam})
```

Comment:
Added WithData at both entries: HandleScale() and GetSCalingHistories(), and removed addtional appid log data